### PR TITLE
USB: samples: Webusb documentation update

### PR DESCRIPTION
**Pull request by finikorg**
Reformat README to rst and update webapp repository.

### CODE REVIEW

File: samples/subsys/usb/webusb/README.rst [hunk 1]
```diff
@@ -1,21 +1,20 @@
-Title: WebUSB Enabled Custom Driver and A Sample Application
+.. _webusb-sample:
 
-This project includes:
+=========================
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: Take a look at http://docs.zephyrproject.org/contribute/doc-guidelines.html for our reST guidelines.

The document title just uses an underline (no overline) with ``#``s and the heading hierarchy after that is ``*``, ``=``, and ``-``.

Change the document title here to be just:

    WebUSB sample application
    #########################

File: samples/subsys/usb/webusb/README.rst [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+WebUSB sample application
+=========================
 
-1. WebUSB Enabled Custom Driver:
---------------------------------
-
-This enables WebUSB support to the device via the platform capability
-descriptors.
+WebUSB Documentation
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: I'd delete this title

File: samples/subsys/usb/webusb/README.rst [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
 
 For a deeper dive into the WebUSB, refer to
 https://github.com/WICG/webusb/blob/gh-pages/explainer.md
 
 WebUSB API Specification:
 https://wicg.github.io/webusb/
 
-2. Sample Application:
-----------------------
+Overview
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: How about:

    Sample Overview
    ***************

File: samples/subsys/usb/webusb/README.rst [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
+--------
 
 A simple echo application to demonstrate the WebUSB enabled custom
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: Make this a complete sentence:

    This simple echo application demonstrates the WebUSB enabled custom class driver.

File: samples/subsys/usb/webusb/README.rst [hunk 5]
```diff
@@ -53,35 +52,37 @@ web page.
 
 Follow below steps to run the demo on host:
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: Change to:

    Follow these steps to run the demo on your host system:

File: samples/subsys/usb/webusb/README.rst [hunk 6]
```diff
@@ -53,35 +52,37 @@ web page.
...
 
-1. Run the latest dev-channel release of Google Chrome on host.
-   https://www.google.com/chrome/browser/desktop/index.html?extra=devchannel
+#. Run the latest Google Chrome on host.
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: Is there a minimum version number needed?  If so, you should mention that.

File: samples/subsys/usb/webusb/README.rst [hunk 7]
```diff
@@ -53,35 +52,37 @@ web page.
...
    To host the demo page locally: Clone the repo and start a web server
    in the appropriate directory.
-   $ python -m SimpleHTTPServer
 
-5. Connect the board to a host.
+   .. code-block:: console
+
+      $ python -m http.server
+
+#. Connect the board to a host.
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: I'd say, ``... to your host.``

File: samples/subsys/usb/webusb/README.rst [hunk 8]
```diff
@@ -53,35 +52,37 @@ web page.
...
-6. Once the device is booted, you should see a notification from
+#. Once the device is booted, you should see a notification from
    Chrome: "Go to localhost to connect.". Click on the notification
    to open demo page.
 
-7. Click on Connect button to connect to the device.
+#. Click on Connect button to connect to the device.
 
-8. Send some text to the device by clicking on Send button. The demo app
+#. Send some text to the device by clicking on Send button. The demo app
```
Comment (dbkinder) [inline samples/subsys/usb/webusb/README.rst]: ``on the Send button.``

**Review by dbkinder — DISMISSED** [review]

File: samples/subsys/usb/webusb/README.rst (hunk 6 above)
Comment (finikorg) [reply samples/subsys/usb/webusb/README.rst]: I do not know from which version Chrome supports this API. At the time of writing Chrome downloaded from official site works.

**Comment by finikorg** [comment]
@dbkinder Updated PR

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**